aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -683,6 +687,17 @@ TORCH_LIBRARY_IMPL(aten, Batched, m) {
 #undef BINARY_POINTWISE_VA
 #undef BINARY_POINTWISE
 
+
+#define TRIVIAL_OP(op) m.impl(#op, \
+    unwrap_and_call<Tensor (*)(const Tensor&), at::op>);
+  // complex number view operators
+  TRIVIAL_OP(imag)
+  TRIVIAL_OP(conj);
+  TRIVIAL_OP(real);
+  TRIVIAL_OP(view_as_complex);
```

Comment:
Will this one work fine if the batch dimension is the last one?
This case is explicitly avoided in the tests with the `shape_postfix` argument.

Reply:
That's a really good point. No, this won't work fine in that case. We'll need a different batching rule here (to move the batch dim to the front).

Reply:
But will this movement of dimension be an issue for view operations? As the move might incur a copy?
Should this just be forbidden? (the last dimension has to be the 2 complex values and cannot be a batch dim)

Reply:
> But will this movement of dimension be an issue for view operations? As the move might incur a copy?

The move won't incur a copy (movement is just a permute).

> Should this just be forbidden? (the last dimension has to be the 2 complex values and cannot be a batch dim)

`view_as_complex` requires that the last dimension has [stride 1 for the view to be valid](https://github.com/pytorch/pytorch/blob/a341a4329a0dd625d890212d72bbd62d3840ba0a/aten/src/ATen/native/ComplexHelper.h#L66) (because we're just viewing [[a0 b0], [a1 b1]] as [a0 b0 a1 b1]. So if the batch dim is at the end of the tensor, we can support `view_as_complex` depending on the strides of the dimension of size 2

```
# This should fail gracefully because the dimension of size 2 doesn't have stride 1.
x = torch.randn(2, B0)
vmap(torch.view_as_complex, in_dim=1)(x)

# This will succeed 
x = torch.randn(B0, 2).t()
vmap(torch.view_as_complex, in_dim=1)(x)